src/app/iteration/iteration.service.ts

```diff
@@ -80,10 +81,35 @@ export class IterationService {
   }
 
   /**
-   * Get locally saved iterations' reference
+   * Update an existing iteration
+   * @param iteration - Updated iteration
+   * @return updated iteration's reference from the list
    */
-  getIterationList() {
-    return this.iterations;
+  updateIteration(iteration: IterationModel): Promise<IterationModel> {
+    return this.http
+      .patch(iteration.links.self, { headers: this.headers })
+      .toPromise()
+      .then (response => {
+        if (response.status != 200) {
+          throw new Error('API error occured');
```

Comment:
I'm not sure how backend looks like - but generally speaking shouldn't we only treat 4xx and 5xx as errors? I'm also not sure if 200 is a correct return code for PATCH - shouldn't it be 204 for example?